project-addons/flask_middleware_connector/models/product/common.py

```diff
@@ -166,10 +166,10 @@ def unlink_product_tag_rel(self):
     def compute_date_next_incoming(self):
         moves = self.env['stock.move'].search(
             [('product_id', '=', self.id), ('purchase_line_id', '!=', False), ('state', 'not in', ['cancel','done']),
-             ('location_dest_id.usage', 'like', 'internal'),('picking_id','!=',False)],
-            limit=1, order="date_expected asc")
+             ('location_dest_id.usage', 'like', 'internal'),'|',('picking_id','!=',False),('container_id','!=',False)]).sorted(
+            key=lambda m: m.date_expected and m.date_reliability)
```

Comment:
Creo que sería más eficiente, poner un store=True con api depends a date_reliability, que serían pocas escrituras y usar el order de sql que es más eficiente y además tal y como estaba, traía un único registro, ahora trae todos los que cumplen la condición (que es posible que no sean muchos, pero bueno) y luego los tiene que ordenar. 
Pero como prefiráis, si se analiza el rendimiento tampoco habrá una diferencia muy grande en este caso, que debería trabajar con pocos registros habitualmente.

Reply:
No deberían ser muchos registros, ya que solo se tienen en cuenta los movimientos de entrada no terminados que o tienen albarán de entrada asociado o contenedor, lo máximo que he llegado a ver a la vez son 3/4 stock_moves. Lo hemos estado hablando y creemos que no va repercutir mucho en el rendimiento tal y como está, y stock_move ya tiene demasiados datos almacenados...  si tu das el visto bueno, lo dejamos así. Si no, lo cambio en un momento.

Reply:
Ok, lo dejamos así